### ISSUE TITLE
`TSForecaster` guide doesn't seem to work

### ISSUE THREAD

**Issue by Lundez** [timeseriesAI/tsai#398]
Hi!

I'm having a problem using `TSForecaster`.
It seems that the performance is equally bad regardless of which model I use, if I use simple `LinearRegression` manually I get a much better results.
Using `darts` with `N-BEATS` or something else also yields much better result, which makes me think that something is bugged.

Code:
```python
from tsai.all import *
import datetime
import pandas_datareader as pdr

crypto_data = pdr.get_data_yahoo(['BTC-USD'], 
                          start=datetime.datetime(2019, 12, 31), 
                          end=datetime.datetime(2020, 12, 31))
btc = crypto_data[('Close', 'BTC-USD')]
X,y = SlidingWindow(window_len=14, horizon=1)(btc)
splits = TimeSplitter(0.2)(y) 
batch_tfms = [TSStandardize(use_single_batch=False)]

fcst = TSForecaster(X, y, splits=splits, batch_tfms=batch_tfms, arch=TST, metrics=mae, bs=512)
fcst.fit_one_cycle(50, 1e-2)

_, y_true, preds = fcst.get_X_preds(X[:2], y[:2])
#(tensor([1887.4600, 1884.5800]), ([513208.75], [517005.5])
```

I have tried using `lr_find`, different architectures, more epochs, more data. Nothing seems to work.
Sometimes I get huge predictions as this case, other times I get minimal predictions (e.g. 0<pred<1)
Also tried on different Stocks (e.g. AMZN)

Is there some kind of scaler missing when you do `get_X_preds`?

**Comment by Lundez** [comment]
@oguiza that's weird. Tried restarting your jupyter session?
Attaching a CSV file with the data.
[btc.csv](https://github.com/timeseriesAI/tsai/files/8056266/btc.csv)

I'd say that it's pretty high. With the current configuration it's atleast < 10k, I've had some configs that where 10-30k.. :/
![image](https://user-images.githubusercontent.com/7490199/153767851-919960a5-75c9-4919-856a-a3b1c7286709.png)

**Comment by oguiza** [comment]
Hi @Lundez,
I have a few comments:

* This is a notoriously difficult task as you know. It's not only difficult to predict financial markets, but you are also training the models in a sideways market and validating on a trending market.
* I'd recommend you to use log returns instead of prices. They are always around zero and they are much easier to predict.  You could use something like this: 

  ```bash
  btc = pd.read_csv('btc.csv').loc[2:, ['Close']].reset_index(drop=True)
  btc['Close'] = btc['Close'].astype(float)
  btc['LogReturn'] = np.log(btc['Close']/ btc['Close'].shift(1))
X,y = SlidingWindow(window_len=14, horizon=1, get_x='LogReturn', get_y='LogReturn')(btc[1:])
  ```

* You can certainly overfit this type of data. It's obviously much more difficult to have a good result in the validation set. This code can take the training metrics to very low values (training mean in 2.23%, so anything lower than that is a good result):
```bash
batch_tfms = TSStandardize(use_single_batch=False)
learn = TSForecaster(X, y, splits=splits, batch_tfms=batch_tfms, arch=InceptionTime, metrics=mae, train_metrics=True, cbs=ShowGraph())
# learn.lr_find()
learn.fit_one_cycle(100, 1e-4) 
```
Let me know if this works for you.

**Comment by Lundez** [comment]
Hi @oguiza thanks for the great tips! 🙏

I tried LogReturn, it did improve the result but not really in the end. Result only got better because distance shown was less, but in reality it wasn't better once reverted into normal values.

I agree it's an incredibly difficult task, and I'm doing this more as a showcase/theory rather than trying to do it for a serious task 😊
As such it's not the full focus of my work-session.

My idea is that this model should be better to forecast the next day based on the 7 previous better than simply taking the value of the day before. Which it doesn't right now.

I'll experiment some more, but it's weird that `darts` seem to result in better prediction with the same data setup. Right?

**Comment by oguiza** [comment]
I'll close this issue due to a lack of response. Please, reopen if necessary.

**Comment by jefferyanderson** [comment]
@Lundez would you mind sharing your darts N-BEATS implementation with me so I can understand how you're doing this? Looking to learn and haven't really grokked darts/N-BEATS yet. Thanks.

**Comment by jefferyanderson** [comment]
@oguiza what is the correct way to use the forecaster to generate n predictions into the future?
My understanding is that learn.get_X_preds will generate one prediction datapoint per input datapoint.

**Comment by Lundez** [comment]
@jefferyanderson @oguiza so sorry for the radio silence, I'll get back in a week or two. 
Currently going through a personal tragedy.

**Comment by oguiza** [comment]
I'm sorry to hear that.  I'll reopen the issue then. Take your time.

**Comment by Lundez** [comment]
I feel like I lost all context and momentum because of the reason. Closing this issue as it seems I'm not gonna find time soon enough to look into this in a reasonable manner.

Thanks for being understanding @oguiza

**Comment by oguiza** [comment]
> @oguiza what is the correct way to use the forecaster to generate n predictions into the future? My understanding is that learn.get_X_preds will generate one prediction datapoint per input datapoint.

You are right. The output will take the same shape as the input y.
